libs/community/langchain_community/callbacks/mlflow_callback.py

```diff
@@ -266,26 +293,21 @@ def __init__(
             experiment_name=self.experiment,
             run_name=self.name,
             run_tags=self.tags,
+            run_id=self.run_id,
+            artifact_dir=self.artifact_dir,
         )
 
         self.action_records: list = []
-        self.nlp = spacy.load("en_core_web_sm")
-
-        self.metrics = {
-            "step": 0,
-            "starts": 0,
-            "ends": 0,
-            "errors": 0,
-            "text_ctr": 0,
-            "chain_starts": 0,
-            "chain_ends": 0,
-            "llm_starts": 0,
-            "llm_ends": 0,
-            "llm_streams": 0,
-            "tool_starts": 0,
-            "tool_ends": 0,
-            "agent_ends": 0,
-        }
+        try:
+            self.nlp = spacy.load("en_core_web_sm")
+        except OSError:
+            logger.warning(
+                "Run `python -m spacy download en_core_web_sm` "
+                "to download en_core_web_sm"
+            )
+            raise
+
+        self.metrics = {key: 0 for key in mlflow_callback_metrics()}
```

Comment:
what does 0 mean here?

Reply:
The original value of all metrics are 0, whenever a certain function is invoked, say on_chain_start, then corresponding metric's value will increase 1 (chain_starts += 1)